### ISSUE TITLE
VoiceOver not reading character strings

### ISSUE THREAD

**Issue by RedEyeDragon** [microsoft/vscode#90840]
@sana-ajani 
@isidorn 

Issue Type: <b>Bug</b>

<h3>Issue</h3>
When moving the cursor across a line of text by word or character VoiceOver is unable to read the text the cursor passes. This basic functionality is essential for in being able to write and review code. 

<h4>Expected:</h4>
When moving across words or character strings VoiceOver should read aloud each character or word the cursor passes. To navigate by character use the left and right arrow keys. Navigating by word use the Option key + left and right arrow keys.

<h4>Actual:</h4>
VoiceOver is not reading the characters or words the cursor passes.

<h4>Recreate:</h4>
<ul>
<li>Turn on VoiceOver</li>
<li>place the cursor on a line containing text</li>
<li>use the left and right arrow keys to move by character</li>
<li>use the Option key + left and right arrow keys to move by word</li>
</ul>

VS Code version: Code 1.42.1 (c47d83b293181d9be64f27ff093689e8e7aed054,  @2020-02-11T14:44:27.652Z)
OS version: Darwin x64 19.3.0

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-6820HQ CPU @ 2.70GHz (8 x 2700)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>protected_video_decode: unavailable_off<br>rasterization: enabled<br>skia_renderer: disabled_off<br>surface_control: disabled_off<br>surface_synchronization: enabled_on<br>video_decode: enabled<br>viz_display_compositor: enabled_on<br>viz_hit_test_surface_layer: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|2, 1, 2|
|Memory (System)|16.00GB (3.72GB free)|
|Process Argv||
|Screen Reader|yes|
|VM|0%|
</details><details><summary>Extensions (3)</summary>

Extension|Author (truncated)|Version
---|---|---
python|ms-|2020.2.63072
python|tht|0.2.3
vscodeintellicode|Vis|1.2.5

</details>

**Comment by isidorn** [comment]
@RedEyeDragon thanks a lot for filling this issue.
This sounds like a potential duplicate of https://github.com/microsoft/vscode/issues/88832
Are you on Mojave or on Catalina. Since on Catalina I can only reproduce the issue I just linked. Other left / right arrow navigation seems to work fine for me.
Under VoiceOver > Verbosity > Text what do you have for "When moving the cursor" setting? Does it help if you set to "Speak text to the right of the cursor"?

**Comment by RedEyeDragon** [comment]
I am on Catalina and I tried it both ways. I have my cursor movement set to speak text cursor passes by default but I switched it and got the same result. I would not say this is a duplicate of the #88832 issue. Although they are similar they are not the same. The previous issue was getting feedback just not correct feedback. This issue could have been fixed with some simple VoiceOver command settings as you suggested. The issue I am documenting is an VS Code editor issue. 

<h2>Example:</h2>
print ("Hello World")
Moving through this string from the left to the right would produce the following feedback after each press of the right arrow key.

"print Hello World"
rint Hello World
int Hello World
nt Hello World
t Hello World
etc

<h3>VoiceOver focus issue?</h3>
I am not sure where the message/notification is located but when ever I move around in the editor it announces details about the python version, and that it is set to accessibility mode. After every arrow key press or movement it does this. Any thoughts?

**Comment by isidorn** [comment]
Ok, since it is a different issue let's keep it open and we can look into  it more next milestone.
The other issue you are hitting is this one https://github.com/microsoft/vscode/issues/87664
It is fixed in Electron 7, thus if you start using vscode insiders you will not longer hit that issue. You can find it here https://code.visualstudio.com/insiders/

**Comment by RedEyeDragon** [comment]
<h2>Issue Resolved in Insider Version!</h2>
I am excited to report that this issue is completely resolved in the insiders release I downloaded! I am able to navigate by line, word, or character and everything is read aloud just as it should be. In my opinion this issues is resolved and can be closed with notes referencing the insiders build.

**Comment by isidorn** [comment]
Great, thanks a lot for letting us know. I will close the issue. And if you hit more issue please open new issues on GitHub and feel fre to ping me on it.
Also please note that next VS Code stable should have new Electron 7, thus you might not need to insiders forever.
